iceoryx_examples/automotive_soa/include/minimal_proxy.hpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_EXAMPLES_AUTOMOTIVE_SOA_MINIMAL_PROXY_HPP
+#define IOX_EXAMPLES_AUTOMOTIVE_SOA_MINIMAL_PROXY_HPP
+
+#include "topic.hpp"
+
+#include "owl/kom/event_subscriber.hpp"
+#include "owl/kom/field_subscriber.hpp"
+#include "owl/kom/method_client.hpp"
+#include "owl/runtime.hpp"
+#include "owl/types.hpp"
+
+/// @note Once a handler has been set with 'EnableFindServiceCallback', calling 'FindService' is not thread-safe!
+class MinimalProxy
+{
+  public:
+    static constexpr char m_serviceIdentifier[] = "MinimalSkeleton";
+
+    MinimalProxy(owl::kom::ProxyHandleType& handle) noexcept;
+    ~MinimalProxy() noexcept = default;
+
+    MinimalProxy(const MinimalProxy&) = delete;
+    MinimalProxy(MinimalProxy&&) = delete;
+    MinimalProxy& operator=(const MinimalProxy&) = delete;
+    MinimalProxy& operator=(MinimalProxy&&) = delete;
+
+    static owl::kom::FindServiceCallbackHandle
+    EnableFindServiceCallback(owl::kom::FindServiceCallback<owl::kom::ProxyHandleType> handler,
```

Comment:
Are we sure this belongs in here? I just ask out of curiosity since this violates in my opinion the separation of concerns.
As far as I understood this the task of a proxy is to receive data? And another object like `ServiceDiscovery` should have the task to find services.
With some API refinement one could also solve the thread-safety issue.

But this isn't something which has to necessarily be fixed, those questions just came up in my mind.

Reply:
Yes, the discovery is part of the proxy class. You can find it [here](https://www.autosar.org//fileadmin/user_upload/standards/adaptive/19-11/AUTOSAR_EXP_ARAComAPI.pdf) on page 24 ff. I agree that from a separation of conerns point of view I don't agree with the API. I think it's connected with the fact that the proxy class is typically generated.